src/test/java/io/jenkins/plugins/analysis/warnings/recorder/TrendCarouselITest.java

```diff
@@ -0,0 +1,271 @@
+package io.jenkins.plugins.analysis.warnings.recorder;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.steps.IssuesRecorder;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.warnings.Java;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.TrendCarousel;
+import io.jenkins.plugins.analysis.warnings.recorder.pageobj.TrendCarousel.TrendChartType;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Provides tests for the trend carousel shown on the details page.
+ *
+ * @author Tobias Redl
+ * @author Andreas Neumeier
+ */
+
+public class TrendCarouselITest extends IntegrationTestWithJenkinsPerSuite {
+
+    private HtmlPage webPage;
+
+    /**
```

Comment:
In unit tests a setup with very small tests would be perfect. In integration tests it makes more sense to combine as many variants as possible in order to speed up the tests. I basically see only one or two big scenarios:
start -> next -> next -> next -> next -> prev -> prev -> prev -> prev -> end and the additional tests for health and local storage.